specs/Standards/MultiToken/Events.md

```diff
@@ -0,0 +1,271 @@
+
+## Summary
+
+Standard interfaces for Multi Token Contract actions.
+
+## Motivation
+
+MT-driven apps such as marketplaces and videogames perform a few 
+core actions `minting`, `burning`, `transferring`, and 
+`approving tokens for transfer`.
+
+Each app has their own way of performing these actions and it
+is difficult to consistently capture these actions. We codify these
+actions as events, in order to capture them from a variety of context
+within the MT Contract. This specification enables the events to be 
+broadly consumed by indexers, developers, and interested systems.
+
+
+Contract implementers are required to emit events for the actions they take. 
+These actions are `minting`,`burning`,`transferring`, and `approving tokens to be transferred`.
+
+This enables indexers, and systems to be able to build a consistent view of the contract and take action on the contract without polling.
+
+Prior Art:
+- [ERC-721]'s events 
+- [ERC-1155]'s events 
+- [NEP-254]'s events
+
+## Events
+
+Many apps use different interfaces that represent the same action.
+This interface standardizes that process by introducing event logs.
+There is no Event NEP yet, so this standard paves the road to that.
+
+Events use standard logs capability of NEAR and defined as a convention.
+Events are log entries that start with `EVENT_JSON:` prefix followed by a single valid JSON document of the following interface:
+
+```ts
+// Interface to capture data
+// about an event
+// Arguments
+// * `standard`: name of standard e.g. nep171
+// * `version`: e.g. 1.0.0
+// * `event`: string
+// * `data`: associate event data
+interface EventLogData {
+    standard: string,
+    version: string,
+    event: string,
+    data?: unknown,
+}
+```
+
+#### Valid event logs:
+
+```js
+EVENT_JSON:{"standard": "nepXXX", "version": "1.0.0", "event": "xyz_is_triggered"}
+```
+
+```js
+EVENT_JSON:{
+  "standard": "nepXXX",
+  "version": "1.0.0",
+  "event": "xyz_is_triggered"
+}
+```
+
+```js
+EVENT_JSON:{"standard": "nepXXX", "version": "1.0.0", "event": "xyz_is_triggered", "data": {"triggered_by": "foundation.near"}}
+```
+
+#### Invalid event logs:
+
+* Two events in a single log entry (instead, call `log` for each individual event)
+```
+EVENT_JSON:{"standard": "nepXXX", "version": "1.0.0", "event": "xyz_is_triggered"}
+EVENT_JSON:{"standard": "nepXXX", "version": "1.0.0", "event": "xyz_is_triggered"}
+```
+* Invalid JSON data
+```
+EVENT_JSON:invalid json
+```
+
+
+## Interface
+Multi Token Events MUST have `standard` set to `"nep246"`, standard version set to `"1.0.0"`, `event` value is one of `mt_mint`, `mt_burn`, `mt_transfer`, `mt_approval` and `data` must be of one of the following relavant types: `MtMintLog[] | MtBurnLog[] | MtTransferLog[] | MtApprovalLog[]`:
+
+
+
+```ts
+// Interface to capture an event. The indexer looks for events prefixed 
+// with EVENT_JSON. The rest follows the EventLog 
+type MtEvent = "mt_mint" | "mt_burn" | "mt_transfer" | "mt_approval"
```

Comment:
```suggestion
type MtEvent = "mt_mint" | "mt_burn" | "mt_transfer"
```

Reply:
I didn't point all these mentions further. We need to make a decision, do we want to have such event log or not,  and then re-check it again